tests/helm/logs_test.go

```diff
@@ -565,3 +565,34 @@ sumologic:
 
 	require.Equal(t, "sumologic|my_logs_namespace", otelConfig.Processors.SourceContainers.Exclude.Namespace)
 }
+
+func TestCollectorDaemonsetUpdateStrategy(t *testing.T) {
+	t.Parallel()
+
+	valuesYaml := `
+otellogs:
+  daemonset:
+    updateStrategy:
+      rollingUpdate:
+        maxUnavailable: 50%
+`
+	templatePath := "templates/logs/collector/otelcol/daemonset.yaml"
+
+	renderedTemplate, err := RenderTemplateFromValuesStringE(t, valuesYaml, templatePath)
+	require.NoError(t, err)
+
+	var logsCollectorDaemonset struct {
+		Spec struct {
+			UpdateStrategy struct {
+				RollingUpdate struct {
+					MaxSurge       string `yaml:"maxSurge"`
+					MaxUnavailable string `yaml:"maxUnavailable"`
+				} `yaml:"rollingUpdate"`
+			} `yaml:"updateStrategy"`
+		}
+	}
+	err = yaml.Unmarshal([]byte(renderedTemplate), &logsCollectorDaemonset)
+	require.NoError(t, err)
+
+	require.Equal(t, "50%", logsCollectorDaemonset.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable)
+}
```

Comment:
Why not goldenfiles?

Reply:
I thought this was the preferred way?

With goldenfiles, it's not clear what we're testing as the test compares the whole file. Also goldenfiles often require updates when something has changed in the template, even when the test doesn't require any changes.